code/modules/antagonists/changeling/powers/absorb.dm

```diff
@@ -56,27 +57,46 @@
 	if(user.nutrition < NUTRITION_LEVEL_WELL_FED)
 		user.set_nutrition(min((user.nutrition + target.nutrition), NUTRITION_LEVEL_WELL_FED))
 
+	if(user.blood_volume < initial(user.blood_volume))
+		user.blood_volume = initial(user.blood_volume)
+
 	if(target.mind)//if the victim has got a mind
 
-		target.mind.show_memory(user, 0) //I can read your mind, kekeke. Output all their notes.
+		var/target_memory = target.mind.show_memory(user, FALSE) //I can read your mind, kekeke. Output all their notes.
 
 		//Some of target's recent speech, so the changeling can attempt to imitate them better.
 		//Recent as opposed to all because rounds tend to have a LOT of text.
 		var/list/recent_speech = list()
-
-		if(target.say_log.len > LING_ABSORB_RECENT_SPEECH)
-			recent_speech = target.say_log.Copy(target.say_log.len-LING_ABSORB_RECENT_SPEECH+1,0) //0 so len-LING_ARS+1 to end of list
+		var/list/target_say_log = list(list(), list())
+		for(var/datum/log_record/log as anything in GLOB.logging.get_logs_by_type(target.last_known_ckey, SAY_LOG))
+			target_say_log[1] += list(list("Chat" = log.what, "Time" = gameTimestamp(wtime = log.raw_time - 9.99)))
+		for(var/datum/log_record/log as anything in GLOB.logging.get_logs_by_type(target.last_known_ckey, EMOTE_LOG))
+			target_say_log[2] += list(list("Chat" = log.what, "Time" = gameTimestamp(wtime = log.raw_time - 9.99)))
+		if(length(target.say_log) > LING_ABSORB_RECENT_SPEECH)
+			recent_speech = target.say_log.Copy(length(target.say_log) - LING_ABSORB_RECENT_SPEECH + 1, 0)
 		else
 			recent_speech = target.say_log.Copy()
 
+		var/account_pin
+		var/account_number
+		if(target.mind.initial_account)
```

Comment:
```suggestion
		if(istype(target.mind.initial_account))
```
type check preferred over null check here

Reply:
if someone gets a zero or another fucked up value in here we kinda want it to fail right?

Reply:
I guess you could go either way, it's unlikely initial_account is gonna have a fucky value like that ever. Feel free to do whatever you think is best, a unit test would be better to catch any issues that would runtime here tbqh.